### PR TITLE
feat: extend doc-build to be windows compatible

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
WIP to extend doc-build action to work with windows and linux.

### CODE REVIEW

**Comment by SMoraisAnsys** [comment]
@RobPasMue Here is a first draft of the modification to allow windows doc build. I need to test it on some repos to ensure that it works correctly but I was able to reach the build command on my fork.

**Comment by RobPasMue** [comment]
Let me try using PyAnsys Geometry with this one =)

**Comment by SMoraisAnsys** [comment]
I'll also try to build pyedb with it using linux and windows.

**Comment by RobPasMue** [comment]
Trying in https://github.com/ansys/pyansys-geometry/pull/1068

File: _doc-build-linux/action.yml [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    description: >
+      Whether to use the Python cache for installing previously downloaded
+      libraries. If ``true``, previously downloaded libraries are installed from the
+      Python cache. If ``false``, libraries are downloaded from the PyPI index.
+    required: false
+    default: true
+    type: boolean
+
+  sphinxopts:
+    description: >
```
Comment (jorgepiloto) [inline _doc-build-linux/action.yml:60]:
```suggestion
    description: |
```

File: doc-build/action.yml
Comment (jorgepiloto) [inline doc-build/action.yml:1]: Could we install Python outside the private actions? This is a common step in both and a cross-platform command.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]

**Pull request closed without merging**